extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxDialog.js

```diff
@@ -0,0 +1,205 @@
+/*
+ * VisualEditor user interface WikiaInfoboxDialog class.
+ */
+
+/**
+ * Dialog for inserting and editing Wikia Infobox templates.
+ *
+ * @class
+ * @extends ve.ui.NodeDialog
+ *
+ * @constructor
+ * @param {Object} [config] Configuration options
+ */
+ve.ui.WikiaInfoboxDialog = function VeUiWikiaInfoboxDialog( config ) {
+	// Parent constructor
+	ve.ui.WikiaInfoboxDialog.super.call( this, config );
+};
+
+/* Inheritance */
+
+OO.inheritClass( ve.ui.WikiaInfoboxDialog, ve.ui.NodeDialog );
+
+/* Static Properties */
+
+ve.ui.WikiaInfoboxDialog.static.name = 'wikiaInfobox';
+
+ve.ui.WikiaInfoboxDialog.static.modelClasses = [ ve.dm.WikiaInfoboxTransclusionBlockNode ];
+
+ve.ui.WikiaInfoboxDialog.static.actions = [
+	{
+		action: 'apply',
+		label: OO.ui.deferMsg( 'visualeditor-dialog-action-apply' ),
+		flags: 'primary'
+	},
+	{
+		label: OO.ui.deferMsg( 'visualeditor-dialog-action-cancel' ),
+		flags: 'safe'
+	}
+];
+
+/* Methods */
+
+/**
+ * @inheritdoc
+ */
+ve.ui.WikiaInfoboxDialog.prototype.getActionProcess = function ( action ) {
+	if ( action === 'apply' ) {
+		return new OO.ui.Process( function () {
+			var surfaceModel = this.getFragment().getSurface();
+			this.transclusionModel.updateTransclusionNode( surfaceModel, this.selectedNode );
+			this.close( { action: action } );
+		}, this );
+	}
+
+	return ve.ui.WikiaInfoboxDialog.super.prototype.getActionProcess.call( this, action );
+};
+
+/**
+ * @inheritdoc
+ */
+ve.ui.WikiaInfoboxDialog.prototype.getSetupProcess = function ( data ) {
+	data = data || {};
+	this.data = data;
+
+	return ve.ui.WikiaInfoboxDialog.super.prototype.getSetupProcess.call( this, data )
+		.next( function () {
+			this.transclusionModel = new ve.dm.WikiaTransclusionModel();
+			this.transclusionModel.setIsInfobox( true );
+
+			// Load existing infobox template
+			this.transclusionModel
+				.load( ve.copy( this.selectedNode.getAttribute( 'mw' ) ) )
```

Comment:
I see that you are manually calling setIsInfobox however I would consider having load(...) method take care of it - after all that's the method that is responsible for actually loading data about the transclusion from the node itself.